### DISCUSSION TITLE
LittleFs on SPI1

### DISCUSSION THREAD

**Discussion by smalltoe** [earlephilhower/arduino-pico#1521] [Q&A] [answered]
Dear All,
On Raspberry Pico I would like to use a flash memory on SPI1. I see this is possible for the [SD Card](https://arduino-pico.readthedocs.io/en/latest/fs.html#using-second-spi-port-for-sd) but I came to the conclusion this argument cannot be passed for LittleFS. So two questions:
1. Is my conclusion correct? (this is the beginner question)
2. Is there a way to still use the flash memory on SPI1?

I have a version that works on Micropython but I would like to port that to Arduino now.

**Comment by earlephilhower** [comment, marked as answer]
You are correct, the `LittleFS` implementation here uses the onboard flash only and does not directly toggle any (Q)SPI pins.

I believe it would be possible to extend the onboard LittleFS by making a new constructor for https://github.com/earlephilhower/arduino-pico/blob/cc800713bd7b30b19ca4b5f02e88f3e915802527/libraries/LittleFS/src/LittleFS.h#L48 and writing replacement flash read/write routines that use a passed-in SPI.  There are function pointers in the struct that are the only thing used by the LittleFS ARM code to acces flash memory.

As for accessing flash via SPI, I suggest you look for some Arduino libraries to handle this or the datasheet of your flash chip for info on the SPI protocol it uses.  It's generally not too complicated to get something low-performance working.

**Reply by smalltoe** [reply]
Many Thanks for confirmation and insights - I will follow down that road then.
